core/pages/CmsPage.js

```diff
@@ -19,19 +21,17 @@ export default {
   watch: {
     '$route': 'validateRoute'
   },
-  asyncData ({ store, route, context }) { // this is for SSR purposes to prefetch data
-    return new Promise((resolve, reject) => {
-      if (context) context.output.cacheTags.add(`cmsPage`)
-      store.dispatch('cmsPage/single', {
+  async asyncData ({ store, route, context }) { // this is for SSR purposes to prefetch data
+    registerModule(CmsModule)
```

Comment:
this needs to be awaited - what if registration will take too long and you will already call `store.dispatch('cmsPage/single'`?